infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -211,14 +210,17 @@ public LogReplicationEntry resend() {
         //Enforce a resend or not
         LogReplicationEntry ack = null;
         boolean force = false;
+        log.info("processAcks {} and pendingMessage size {}", pendingCompletableFutureForAcks.keySet(), pendingMessages.getSize());
+
+        if (pendingMessages.isEmpty()) {
+            return null;
+        }
+
         try {
             ack = processAcks();
-        } catch (TimeoutException e) {
+        } catch (Exception e) {
```

Comment:
Then change the logging message which indicates it is a timeout exception

Reply:
The processAcks can have other exceptions.